### PR TITLE
POAP as reward

### PR DESCRIPTION
**Pull request by BrickheadJohnny**

### CODE REVIEW

File: src/requirements/Payment/PaymentRequirement.tsx [hunk 1]
```diff
@@ -79,8 +73,7 @@ const PaymentRequirement = (props: RequirementProps): JSX.Element => {
       rightElement={
         props?.rightElement ? (
           <GuildCheckoutProvider>
-            {(satisfiesRequirement && !multiplePayments) ||
-            (poapLinks && poapLinks?.claimed === poapLinks?.total) ? (
+            {satisfiesRequirement && !multiplePayments ? (
```
Comment (BrickheadJohnny) [inline src/requirements/Payment/PaymentRequirement.tsx:76]: Deleted the mint link availability check here, I'm not sure if we need that or not? What do you think?
In my opinion, if we decide to include a check, we should check if the role has any text-based rewards (NFT, Secret, Unique secret, POAP) & if it has, we should disable the pay button, but still not sure if we need that extra logic here or not. Lmk what do you think!

File: src/platforms/Poap/PoapCardButton.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const { isAdmin } = useGuildPermission()
+
+  const rolePlatform = roles
+    ?.find((r) => r.rolePlatforms.some((rp) => rp.guildPlatformId === platform.id))
+    ?.rolePlatforms?.find((rp) => rp.guildPlatformId === platform?.id)
+
+  const startTimeDiff = getTimeDiff(rolePlatform?.startTime)
+  const endTimeDiff = getTimeDiff(rolePlatform?.endTime)
+
+  const isButtonDisabled =
```
Comment (BrickheadJohnny) [inline src/platforms/Poap/PoapCardButton.tsx:25]: This logic is repeated in multiple places (NFT, text rewards, POAP), but didn't want to add more changes to this PR, so I made a Linear issue, will make sure to make a hook for this. 👍

File: src/components/[guild]/collect/components/CollectNft.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+const availibiltyTagStyleProps = {
+  bgColor: "transparent",
+  fontSize: "sm",
+  fontWeight: "medium",
+  colorScheme: "purple",
+  p: 0,
+}
+
+const CollectNft = () => {
```
Comment (BrickheadJohnny) [inline src/components/[guild]/collect/components/CollectNft.tsx:33]: No need to review this one, it's just a copy-paste from the old RequirementsCard component

File: src/components/[guild]/claim-poap/components/ClaimPoapButton.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+  const { urlName, roles } = useGuild()
+  const { address } = useAccount()
+
+  const roleId = roles?.find((role) =>
+    role.rolePlatforms.some((rp) => rp.id === rolePlatformId)
+  )?.id
+  const { isLoading: isAccessLoading, hasAccess } = useAccess(roleId)
+
+  // TODO: we'll be able to fetch this from our API once PR#1011 is merged
```
Comment (BrickheadJohnny) [inline src/components/[guild]/claim-poap/components/ClaimPoapButton.tsx:41]: Made a linear issue for this, I'll be notified once that PR is merged 👌

File: src/types.ts [hunk 5]
```diff
@@ -205,6 +205,8 @@ type PlatformGuildData = {
     text?: never
     texts?: never
     imageUrl?: never
+    fancyId?: never
```
[the remark this answers is not in the document]
Comment (BrickheadJohnny) [reply src/types.ts:208]: We'll have a proper type & schema for this in our types package, this was a quick & dirty solution ;)

**Review by dovalid — APPROVED** [review]

**Pull request merged**